### PR TITLE
Us41 password verification

### PR DESCRIPTION
**Pull request by AnnesophieMarchet**

### CODE REVIEW

**Review by wilderLPM — APPROVED** [review]
ok

File: client/src/components/Form/SignUpForm.jsx [hunk 1]
```diff
@@ -41,100 +70,126 @@ function SignUpForm() {
...
             id="confirmPassword"
             name="confirmPassword"
             value={register.confirmPassword}
             onChange={handleUpdateForm}
-            className="w-full px-3 py-2 border rounded focus:outline-none focus:border-secondary focus:ring focus:ring-primary focus:ring-opacity-20"
+            className={`${inputStyles}
+              ${errors.confirmPassword ? "border-red-500" : ""}`}
           />
+
+          {errors.confirmPassword && (
```
Comment (wildmatthieu) [inline client/src/components/Form/SignUpForm.jsx:190]: explicitez la condition : 
`errors.confirmPassword === ??? &&`

File: client/src/components/Form/SignUpForm.jsx [hunk 2]
```diff
@@ -41,100 +70,126 @@ function SignUpForm() {
...
+            aria-describedby="passwordDescribe"
             type="password"
             id="password"
             name="password"
             value={register.password}
             onChange={handleUpdateForm}
-            className="w-full px-3 py-2 border rounded focus:outline-none focus:border-secondary focus:ring focus:ring-primary focus:ring-opacity-20"
+            className={`${inputStyles}  ${errors.password ? "border-red-500" : ""}`}
           />
+          {errors.password && <p className="text-red-500">{errors.password}</p>}
```
Comment (wildmatthieu) [inline client/src/components/Form/SignUpForm.jsx:172]: explicitez la condition : `errors.password === ??? && ...`

File: client/src/components/Form/SignUpForm.jsx [hunk 3]
```diff
@@ -41,100 +70,126 @@ function SignUpForm() {
...
             value={register.sex}
             onChange={handleUpdateForm}
-            className="w-full px-3 py-2 border rounded focus:outline-none focus:border-secondary focus:ring focus:ring-primary focus:ring-opacity-20"
+            className={`${inputStyles} ${errors.sex ? "border-red-500" : ""}`}
           >
+            <option value="femme">{}</option>
             <option value="femme">Féminin</option>
             <option value="homme">Masculin</option>
           </select>
+          {errors.sex && <p className="text-red-500">{errors.sex}</p>}
```
Comment (wildmatthieu) [inline client/src/components/Form/SignUpForm.jsx:155]: explicitez la condition : `errors.sex === ??? && ...`

File: client/src/components/Form/SignUpForm.jsx [hunk 4]
```diff
@@ -41,100 +70,126 @@ function SignUpForm() {
...
             name="birthday"
             value={register.birthday}
             onChange={handleUpdateForm}
-            className="w-full px-3 py-2 border rounded focus:outline-none focus:border-secondary focus:ring focus:ring-primary focus:ring-opacity-20"
+            className={`${inputStyles}${
+              errors.birthday ? "border-red-500" : ""
+            }`}
           />
         </div>
+        {errors.birthday && <p className="text-red-500">{errors.birthday}</p>}
```
Comment (wildmatthieu) [inline client/src/components/Form/SignUpForm.jsx:137]: explicitez la condition : `errors.birthday == ??? && ...`

File: client/src/components/Form/SignUpForm.jsx [hunk 5]
```diff
@@ -41,100 +70,126 @@ function SignUpForm() {
...
             type="text"
             id="firstname"
             name="firstname"
             value={register.firstname}
             onChange={handleUpdateForm}
-            className="w-full px-3 py-2 border rounded focus:outline-none focus:border-secondary focus:ring focus:ring-primary focus:ring-opacity-20"
+            className={`${inputStyles} ${errors.firstname ? "border-red-500" : ""}`}
           />
+
+          {errors.firstname && (
```
Comment (wildmatthieu) [inline client/src/components/Form/SignUpForm.jsx:82]: explicitez la condition : `errors.firstname === ??? && ...`

File: client/src/components/Form/SignUpForm.jsx [hunk 6]
```diff
@@ -41,100 +70,126 @@ function SignUpForm() {
...
             id="lastname"
             name="lastname"
             value={register.lastname}
             onChange={handleUpdateForm}
-            className="w-full px-3 py-2 border rounded focus:outline-none focus:border-secondary focus:ring focus:ring-primary focus:ring-opacity-20"
+            className={`${inputStyles} ${
+              errors.lastname ? "border-red-500" : ""
+            }`}
           />
+          {errors.lastname && <p className="text-red-500">{errors.lastname}</p>}
```
Comment (wildmatthieu) [inline client/src/components/Form/SignUpForm.jsx:102]: explicitez la condition : `errors.lastname === ??? && ...`

File: client/src/components/Form/SignUpForm.jsx [hunk 7]
```diff
@@ -41,100 +70,126 @@ function SignUpForm() {
...
             aria-required="true"
+            aria-describedby="lastnameDescribe"
             type="text"
             id="lastname"
             name="lastname"
             value={register.lastname}
             onChange={handleUpdateForm}
-            className="w-full px-3 py-2 border rounded focus:outline-none focus:border-secondary focus:ring focus:ring-primary focus:ring-opacity-20"
+            className={`${inputStyles} ${
+              errors.lastname ? "border-red-500" : ""
```
Comment (wildmatthieu) [inline client/src/components/Form/SignUpForm.jsx:99]: explicitez la condition : `errors.lastname === ??? && ...`

File: client/src/components/Form/SignUpForm.jsx [hunk 8]
```diff
@@ -41,100 +70,126 @@ function SignUpForm() {
...
+            autoComplete="password"
             aria-required="true"
+            aria-describedby="passwordDescribe"
             type="password"
             id="password"
             name="password"
             value={register.password}
             onChange={handleUpdateForm}
-            className="w-full px-3 py-2 border rounded focus:outline-none focus:border-secondary focus:ring focus:ring-primary focus:ring-opacity-20"
+            className={`${inputStyles}  ${errors.password ? "border-red-500" : ""}`}
```
Comment (wildmatthieu) [inline client/src/components/Form/SignUpForm.jsx:170]: explicitez la condition : `errors.password === ??? ...`

File: client/src/components/Form/SignUpForm.jsx [hunk 9]
```diff
@@ -41,100 +70,126 @@ function SignUpForm() {
...
             aria-required="true"
+            aria-describedby="confirmPasswordDescribe"
             type="password"
             id="confirmPassword"
             name="confirmPassword"
             value={register.confirmPassword}
             onChange={handleUpdateForm}
-            className="w-full px-3 py-2 border rounded focus:outline-none focus:border-secondary focus:ring focus:ring-primary focus:ring-opacity-20"
+            className={`${inputStyles}
+              ${errors.confirmPassword ? "border-red-500" : ""}`}
```
Comment (wildmatthieu) [inline client/src/components/Form/SignUpForm.jsx:187]: explicitez la condition : `errors.confirmPassword === ??? ...`

**Review by wildmatthieu — CHANGES_REQUESTED** [review]

**Pull request merged**